aten/src/ATen/native/sparse/SparseBinaryOpIntersectionStubs.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <ATen/core/Tensor.h>
+#include <ATen/native/DispatchStub.h>
+
+namespace at {
+namespace native {
+
+using mul_sparse_sparse_out_fn = void (*)(Tensor&, const Tensor&, const Tensor&);
+
+DECLARE_DISPATCH(mul_sparse_sparse_out_fn, mul_sparse_sparse_out_stub);
```

Comment:
You are missing the `DEFINE_DISPATCH` call?

Reply:
Define dispatch takes place where it is being used, or am I wrong? I also find the usage of define/declare confusing in such a context.

Reply:
dispatch through native also seems to fail :(

Reply:
> Define dispatch takes place where it is being used, or am I wrong?

No, `DEFINE_DISPATCH` must exist in exactly one file and it must be within `libtorch_cpu.so`. Usually this would be in the file that calls the dispatch stub.

> I also find the usage of define/declare confusing in such a context.

If you look at `DispatchStub.h` you can see the stub is simply an `extern` variable. So `DECLARE_DISPATCH` is the header declaration `extern MyStruct foo`, `DEFINE_DISPATCH` is the cpp file definition `MyStruct foo`.